client/state/data-layer/handler-loading.js

```diff
@@ -0,0 +1,28 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import { mergeHandlers } from 'state/action-watchers/utils';
+
+export const state = {
+	handlers: {},
+};
```

Comment:
What if the `state` variable was called `registeredHandlers` instead? Right now, I have to think very hard to understand the code: is `state` the Redux state object, like everywhere else, or something different? The name doesn't help me at all.

Also, what if the `registeredHandlers` was a mutable object without the `handlers` subproperty, and `registerHandlers` just merged new properties into it? The code that updates the registered handlers seems unnecessarily complicated. For example, why is there the `if ( state.handlers !== nextHandlers )` check? Does it prevent anything bad from happening?

Of course, an ideal API for this module would be to export just two functions: `registerHandlers` and `findHandlers` and keep the actual data structure private to the module. Then we can modify the structure however we want (possibly to allow unregistration) without changing the interface.

These are just optional, although very desirable, improvements. The PR will work very well even without them.

Reply:
excellent suggestions @jsnajdr - in fact I think these are evidence of the experimental nature of how I wrote this. I like your feedback so well I'm going to go ahead and make these changes. internalizing the abstraction is A+++

it wasn't until I discovered that a circular dependency was behind a nasty bug that I split out the registration function into a separate module. artifacts all over…